specs/pvs/corla.pvs

```diff
@@ -2800,23 +2803,112 @@ BEGIN
+%%% \texttt{upload\_\-sucessful}, \texttt{checking\_hash},
+%%% \texttt{hash\_wrong}, \texttt{hash\_verified},
+%%% \texttt{parsing\_\-data}, \texttt{file\_\-type\_\-wrong}, and
+%%% \texttt{data\_parsed}.
+    (: "ballot_manifest-status", "cvr-status" :)
+%%% There are several possible cases where a client can make an
+%%% inappropriate call to the server. Our client will be guarnateed to
+%%% never make sure a call, but we cannot presume that only our client
+%%% will be communicating with the server, as the server will be
+%%% publicly accessible on the internet. We call these behaviors of
+%%% the server ``exceptional behavior''. All responses to exceptional
+%%% behavior come in the form of HTTP client error (\texttt{3xx}) or
+%%% server error (\texttt{4xx}) response codes.
+  http_response_code: TYPE = {
+    OK_200,
+    BAD_REQUEST_400,
+    UNAUTHORIZED_401,
+    FORBIDDEN_403,
+    NOT_FOUND_404,
+    UNSUPPORTED_MEDIA_TYPE_415,
+    UNPROCESSABLE_ENTITY_422,
+    INTERNAL_SERVER_ERROR_500
+  }
+%%% The cases that we must cover are:
+%%% \begin{enumerate}
+%%%
+%%% \item The server is simply not up and running properly. For
+%%% example, if the database is not in a legal state. In these cases
+%%% the server will answer with an HTTP response code of
+%%% \texttt{INTERNAL\_SERVER\_ERROR\_500}.
+%%%
+%%% \item In instances where the server is asked to query for a
+%%% particular piece of data from the database and that does does not
+%%% exist, the server will respond with a \texttt{NOT\_FOUND\_404}. This
+%%% can happen, e.g., when a client uses an improper encoding for a
+%%% piece of data, such as a county ID that does not exist.
+%%%
+%%% \item A piece of data uploaded by a client (at this point in time,
+%%% either a ballot manifest, a CVR file, or an Audit CVR) is
+%%% malformed and cannot be interpreted. This situation can happen,
+%%% for example, if the Election Management System used to export the
+%%% file has a bug, if the file is improperly edited by someone, or if
+%%% there is a bug in our parser. In this situation, the server will
+%%% respond with a \texttt{UNPROCESSABLE\_ENTITY\_422}
+%%% response.
+%%%
+%%% \item Alternatively, if the file is of the wrong type (i.e., a PDF
+%%% was uploaded when the server expected a CSV file), or if the
+%%% uploaded file is too large, then the server will respond with a
+%%% \texttt{UNSUPPORTED\_MEDIA\_TYPE\_415} message.
+%%%
+%%% \item If the client is unauthenticated, or if authentication
+%%% fails, then the server will respond with a
+%%% \texttt{UNAUTHORIZED\_401} response.
+%%%
+%%% \item If some element of a request is incorrect then the server
+%%% responds with a \texttt{BAD\_REQUEST\_400} messages. In particular,
+%%% if the digest uploaded with a ballot manifest or a CVR file is
+%%% incorrect, then the server will store the data and respond with
+%%% this message.
+%%%
+%%% \item Finally, if the endpoint called is improper given the state
```

Comment:
Maybe clarify what "improper" means via e.g.: "if calling the given endpoint implies an illegal state transition,"